ansible_base/authenticator_plugins/saml.py

```diff
@@ -0,0 +1,191 @@
+            " documentation for additional details and syntax."
+        ),
+    )
+    SP_EXTRA = serializers.JSONField(
+        default={"requestedAuthnContext": False},
+        help_text=_("A dict of key value pairs to be passed to the underlying python-saml Service Provider configuration setting."),
+    )
+    SECURITY_CONFIG = serializers.JSONField(
+        default={},
+        help_text=_(
+            "A dict of key value pairs that are passed to the underlying python-saml security setting https://github.com/onelogin/python-saml#settings"
+        ),
+    )
+    EXTRA_DATA = serializers.ListField(
+        default=[],
+        help_text=_("A list of tuples that maps IDP attributes to extra_attributes. Each attribute will be a list of values, even if only 1 value."),
+    )
+    IDP_URL = URLField(
+        allow_null=False,
+        help_text=_("The URL to redirect the user to for login initiation."),
+    )
+    IDP_X509_CERT = PublicCert(
+        allow_null=False,
+        help_text=_("The public cert used for secrets coming from the IdP."),
+    )
+    IDP_ENTITY_ID = serializers.CharField(
+        allow_null=False,
+        help_text=_("The entity ID returned in the assertion."),
+    )
+    IDP_GROUPS = serializers.CharField(
+        help_text=_("The field in the assertion which represents the users groups."),
+    )
+    IDP_ATTR_EMAIL = serializers.CharField(
+        allow_null=False,
+        help_text=_("The field in the assertion which represents the users email."),
+    )
+    IDP_ATTR_USERNAME = serializers.CharField(
+        help_text=_("The field in the assertion which represents the users username."),
+    )
+    IDP_ATTR_LAST_NAME = serializers.CharField(
+        allow_null=False,
+        help_text=_("The field in the assertion which represents the users last name."),
+    )
+    IDP_ATTR_FIRST_NAME = serializers.CharField(
+        allow_null=False,
+        help_text=_("The field in the assertion which represents the users first name."),
+    )
+    IDP_ATTR_USER_PERMANENT_ID = serializers.CharField(
+        help_text=_("The field in the assertion which represents the users permanent id (overrides IDP_ATTR_USERNAME)"),
+    )
+
+    def validate(self, attrs):
+        # attrs is only the data in the configuration field
+        errors = {}
+        # pull the cert_info out of the existing object (if we have one)
+        cert_info = {
+            "SP_PRIVATE_KEY": getattr(self.instance, 'configuration', {}).get('SP_PRIVATE_KEY', None),
+            "SP_PUBLIC_CERT": getattr(self.instance, 'configuration', {}).get('SP_PUBLIC_CERT', None),
+        }
+
+        # Now get the cert_info out of the passed in attrs (if there is any)
+        for cert_type in cert_info.keys():
+            item = attrs.get(cert_type, None)
+            if item and item == ENCRYPTED_STRING and not self.instance:
```

Comment:
But `SP_PUBLIC_CERT` isn't encrypted, right? At least, it's not in `configuration_encrypted_fields` below, so I think we can avoid looping here and just check `SP_PRIVATE_KEY`?

Reply:
Yes, initially I was doing more in this loop but since that got moved out we can just check the PRIVATE_KEY.